### PR TITLE
emphasize cuda kernel functions are asynchronous

### PR DESCRIPTION
**Pull request by yumasheta**
resolves numba#5992

Rephrase part of cuda kernel docs to emphasize that cuda kernels are asynchronous per default.

### CODE REVIEW

**Comment by esc** [comment]
@yumasheta thanks for submitting this, I have added it to the queue for review.

**Comment by stuartarchibald** [comment]
@gmarkall could you take a look at this please? Thanks.

**Comment by yumasheta** [comment]
@gmarkall thanks for the suggestions. I added a word here and there for more clarity.
Also I noticed the wrong link target for cuda.synchronize() (sry for that) and fixed it. Required a new section heading in the cuda-reference/host.rst file.

File: docs/source/cuda/kernels.rst [hunk 1]
```diff
@@ -73,9 +73,15 @@ We notice two steps here:
   (here ``increment_by_one``) and indexing it with a tuple of integers.
 
 * Running the kernel, by passing it the input array (and any separate
-  output arrays if necessary).  By default, running a kernel is synchronous:
-  the function returns when the kernel has finished executing and the
-  data is synchronized back.
+  output arrays if necessary). Kernels run asynchronously: launches queue their
+  execution on the device and then return immediately.  You can use 
+  :ref:`cuda.synchronize() <cuda-host-context-management>` to wait for all previous
+  kernel launches to finish executing.
```
[the remark this answers is not in the document]
Comment (yumasheta) [reply docs/source/cuda/kernels.rst:79]: sweet! I didn't know that! I'll fix the link.
The other commit on the cuda-reference/host.rst file is now obsolete.
What do you want me to do with that? If anything...

[the remark this answers is not in the document]
Comment (yumasheta) [reply docs/source/cuda/kernels.rst:79]: I'll remove that commit from the update-cuda-kernel-docs branch.

**Pull request merged**